ansible/roles_ocp_workloads/ocp4_workload_migration_training/tasks/remove_workload.yml

```diff
@@ -0,0 +1,92 @@
+---
+# Implement your Workload removal tasks here
+
+- name: Remove MTC
+  when: ocp4_workload_migration_training_install_mtc | bool
+  block:
+  - name: Remove Migration Controller
+    k8s:
+      state: absent
+      api_version: migration.openshift.io/v1alpha1
+      kind: MigrationController
+      name: migration-controller
+      namespace: openshift-migration
+
+  - name: Get Installed CSVs
+    k8s_info:
+      api_version: operators.coreos.com/v1alpha1
+      kind: ClusterServiceVersion
+      namespace: openshift-migration
+    register: r_csvs
+
+  - name: Set CSV Name
+    when: r_csvs.resources | length > 0
+    set_fact:
+      __ocp4_workload_migration_training_csv_plan_name: "{{ r_csvs.resources | to_json | from_json | json_query(query) }}"
+    vars:
+      query: >-
+        [?starts_with(metadata.name, 'mtc-operator')].metadata.name|[0]
+
+  - name: Remove CSV
+    when: __ocp4_workload_migration_training_csv_plan_name is defined
+    k8s:
+      state: absent
+      api_version: operators.coreos.com/v1alpha1
+      kind: ClusterServiceVersion
+      name: "{{ __ocp4_workload_migration_training_csv_plan_name }}"
+      namespace: openshift-migration
+
+  - name: Find InstallPlans
+    k8s_info:
+      api_version: operators.coreos.com/v1alpha1
+      kind: InstallPlan
+      namespace: openshift-migration
+    register: r_install_plans
+
+  - name: Set InstallPlan Name
+    when: r_install_plans.resources | length > 0
+    set_fact:
+      ocp4_workload_migration_training_install_plan_name: "{{ r_install_plans.resources | to_json | from_json | json_query(query) }}"
+    vars:
+      query: >-
+        [?starts_with(spec.clusterServiceVersionNames[0], 'mtc-operator')].metadata.name|[0]
+
+  - name: Remove InstallPlan
+    when:
+    - ocp4_workload_migration_training_install_plan_name is defined
+    - ocp4_workload_migration_training_install_plan_name |default("") | length > 0
+    k8s:
+      state: absent
+      api_version: operators.coreos.com/v1alpha1
+      kind: InstallPlan
+      name: "{{ ocp4_workload_migration_training_install_plan_name }}"
+      namespace: openshift-migration
+
+- name: Remove CatalogSource for catalog snapshot
```

Comment:
I suppose this works either way, but is there a reason you wouldn't want a conditional here looking at `cp4_workload_migration_training_use_catalog_snapshot` ?

Reply:
If it doesn't exist than this is no-op